common/vts.proto

```diff
@@ -11,109 +11,52 @@ import "attestation.proto";
 import "token.proto";
 
 
-enum TAType {
-    TA_UNSPECIFIED = 0;
-    TA_RAWPUBLICKEY = 1;
-    // Refer https://github.com/veraison/veraison/issues/76, to introduce certs here!
-    TA_INVALID = 255;
-}
-
-message EndorsementID {
-    AttestationFormat type = 1;
-    google.protobuf.Struct parts = 2;
-}
-
-message Evidence {
-    google.protobuf.Struct value = 1;
-}
-
-message GetEndorsementsRequest{
-    EndorsementID id = 1;
-    Evidence evidence = 2;
-}
-
 message Status {
     bool result = 1;
     string error_detail = 2;
 }
 
-message GetEndorsementsResponse {
-    Status status = 1;
-    string details = 2;
-    TrustVector trust_vector = 3;
-    google.protobuf.Struct endorsements = 4;
-}
-
-message SwComponentID {
-    AttestationFormat type = 1;
-    google.protobuf.Struct parts = 2;
-}
-
-message SwComponent {
-    SwComponentID id = 1;
-    google.protobuf.Struct attributes = 2;
+message Evidence {
+    google.protobuf.Struct value = 1;
 }
 
 message AddSwComponentsRequest {
-    repeated SwComponent info = 1;
+    repeated Endorsement sw_components = 1;
 }
 
 message AddSwComponentsResponse {
     Status status = 1;
 }
 
-message GetSwComponentRequest {
-    SwComponentID id = 1;
-}
-
-message GetSwComponentResponse {
-    Status status = 1;
-    google.protobuf.Struct attributes = 2;
-}
-
-message TrustAnchorID {
-    AttestationFormat type = 1;
-    google.protobuf.Struct parts = 2;
+enum EndorsementType {
+    UNSET = 0;
+    REFERENCE_VALUE = 1;
+    VERIFICATION_KEY = 2;
 }
 
-message TrustAnchorValue {
-    TAType type = 1;
-    google.protobuf.Struct value = 2;
-}
-
-message TrustAnchor {
-    TrustAnchorID id = 1;
-    TrustAnchorValue value = 2;
+message Endorsement {
+    AttestationFormat scheme = 1;
+    EndorsementType type = 2;
+    google.protobuf.Struct attributes = 3;
 }
 
 message AddTrustAnchorRequest {
-    TrustAnchor trust_anchor = 1;
+    Endorsement trust_anchor = 1;
 }
 
 message AddTrustAnchorResponse {
     Status status = 1;
 }
 
-message GetTrustAnchorRequest {
-    TrustAnchorID ta_id = 1;
-}
-
-message GetTrustAnchorResponse {
-    Status status = 1;
-    TrustAnchorValue value = 2;
-}
-
 // Client interface for the Veraison Trusted Services component.
 service VTS {
     // Returns attestation information -- evidences, endorsed claims, trust
     // vector, etc -- for the provided attestation token data.
     rpc GetAttestation(AttestationToken) returns (Attestation);
 
-    // Service endpoints that are used to store and retrieve Software Components
-    // and Trust Anchors to and from the endorsement store
+    // Service endpoints that are used to store Software Components
```

Comment:
If Trust Anchors and SW Component elements are in itself Endorsement as per your proposal, what is the benefit of retaining these two messages ?

I can just have one message:

AddEndorsements(EndrosementRequest) 
==> Type is enough to qualify what EndorsementType is been added ???

Reply:
The current store buckets are Endorsement Store and TrustAnchor store, so how this is aligned ?

Reply:
The contract with the store layer is that SwComponents go to the Endorsements store while any crypto material goes into the TrustAnchor store.

Reply:
It's tricky.  Make a naming proposal maybe?  And we can resolve the naming issue in a different PR.

Reply:
We need to be consistent to call out the store names to be aligned with the interface! 

Better to remove the name Endorsement Store and call it REFVAL Store 
Then it becomes neater and consistent!

Otherwise it is a mess ! We call both RefVal and TA as Endorsement but does not keep TA inside Endorsement Store ???

Reply:
> Better to remove the name Endorsement Store and call it REFVAL Store Then it becomes neater and consistent!

I am not sure.  I think the same store could end up containing "endorsed values" (e.g., security certification status).

> Otherwise it is a mess ! We call both RefVal and TA as Endorsement but does not keep TA inside Endorsement Store ???

As I said, naming this stuff is a tricky thing, mainly because the existing terminology is quite lacking.

Reply:
Veraison needs to be able to distinguish cryptographic endorsements (which may need a different treatment in terms of storage security guarantees) from the rest.

Reply:
> > Better to remove the name Endorsement Store and call it REFVAL Store Then it becomes neater and consistent!
> 
> I am not sure. I think the same store could end up containing "endorsed values" (e.g., security certification status) alongside "reference values".
>> Why? We could have endorsed-vals store ?
> 
> > Otherwise it is a mess ! We call both RefVal and TA as Endorsement but does not keep TA inside Endorsement Store ???
> 
> As I said, naming this stuff is a tricky thing, mainly because the existing terminology is quite lacking.

Reply:
how many stores do we want to maintain? :-)

Reply:
It is a serious question though: I think having a "cryptographic endorsements" and "normal endorsements" should be enough.